### PR TITLE
Added logging if transaction does not passed signature check condition IR-1168

### PR DESCRIPTION
**Pull request by rkharisov**
Signed-off-by: rkharisov <rinat@soramitsu.co.jp>

### Description of the Change
Previously if signature in transaction did nod pass condition described in account, then nothing was logged and there is no way to define if something wrong. Such situation can occur, for example, when transaction was signed with key that was not associated with the account

### Benefits
A little help to define what happened

### Possible Drawbacks 
Additional computation during handling transactions ready to be comitted

### CODE REVIEW

File: iroha/src/queue.rs [hunk 1]
```diff
@@ -113,7 +115,7 @@ impl<W: WorldTrait> Handler<GetPendingTransactions> for Queue<W> {
     type Result = PendingTransactions;
     async fn handle(
         &mut self,
-        GetPendingTransactions: GetPendingTransactions,
+        _get_pending_transactions: GetPendingTransactions,
```
Comment (i1i1) [inline iroha/src/queue.rs:118]: It was intentional. Previously there was deconstruction of zero size structure. It should break after addition of fields to it. So it is better to have it as it was before

**Review by i1i1 — CHANGES_REQUESTED** [review]

**Comment by i1i1** [comment]
Dont feel like we need to add some code complexity for preety printing

File: iroha/src/queue.rs (hunk 1 above)
Comment (Revertron) [reply iroha/src/queue.rs:118]: I think we need a comment about that there ;)

Comment (rkharisov) [reply iroha/src/queue.rs]: reverted

**Comment by rkharisov** [comment]
> Dont feel like we need to add some code complexity for preety printing

Actually, I agree. But as far as I know the way how signature condition check and further logic will be changed soon, so current solution is kind a temporal workaround. @eadventurous What do you think?

**Comment by eadventurous** [comment]
> > Dont feel like we need to add some code complexity for preety printing
> 
> Actually, I agree. But as far as I know the way how signature condition check and further logic will be changed soon, so current solution is kind a temporal workaround. @eadventurous What do you think?

To my mind it is okay to add this workaround here. As this is really a huge problem for client side developers.

File: iroha/src/queue.rs (hunk 1 above)
Comment (i1i1) [reply iroha/src/queue.rs]: > I think we need a comment about that there ;)
This is kinda silly to comment some language constructs

**Comment by i1i1** [comment]
> > > Dont feel like we need to add some code complexity for preety printing
> > 
> > 
> > Actually, I agree. But as far as I know the way how signature condition check and further logic will be changed soon, so current solution is kind a temporal workaround. @eadventurous What do you think?
> 
> To my mind it is okay to add this workaround here. As this is really a huge problem for client side developers.

May be we can add some websocket events for that? Like for dropping things via ttl

**Comment by rkharisov** [comment]
> > > > Dont feel like we need to add some code complexity for preety printing
> > > 
> > > 
> > > Actually, I agree. But as far as I know the way how signature condition check and further logic will be changed soon, so current solution is kind a temporal workaround. @eadventurous What do you think?
> > 
> > 
> > To my mind it is okay to add this workaround here. As this is really a huge problem for client side developers.
> 
> May be we can add some websocket events for that? Like for dropping things via ttl

I think it is possible. But again, if there will be soon changes in logic of processing signature check condition in MST, it makes all affords to sending event to client probably worthless

File: iroha/src/queue.rs [hunk 2]
```diff
@@ -197,6 +200,13 @@ impl<W: WorldTrait> Queue<W> {
                 }
             };
 
+            if !signature_condition_passed && self.signature_condition_not_passed.insert(tx_hash) {
+                iroha_logger::debug!(
```
Comment (eadventurous) [inline iroha/src/queue.rs:204]:
```suggestion
                iroha_logger::warn!(
```

**Comment by eadventurous** [comment]
Dropping because of ttl is a bit of a separate matter, which we can also log of course. Unsatisfied signature condition being not satisfied until ttl - will lead to transaction being dropped. But also just reporting unsatisfied signature condition can be also useful.

**Comment by i1i1** [comment]
> Dropping because of ttl is a bit of a separate matter, which we can also log of course. Unsatisfied signature condition being not satisfied until ttl - will lead to transaction being dropped. But also just reporting unsatisfied signature condition can be also useful.

Ikr, that is why logging only ttl should solve issue

**Comment by eadventurous** [comment]
It might be too long to wait for the TTL.

**Comment by i1i1** [comment]
> It might be too long to wait for the TTL.

Yeah, but you wont be sure otherwise. Signature check is custom predicate with its own fsm. Should we log it also? I dont think so.

**Comment by eadventurous** [comment]
Egor Ivkov, [06.07.21 13:10]
[In reply to Ivan Rybin]
Ok, let's log drop due to TTL for now

Egor Ivkov, [06.07.21 13:10]
With a warn log level

**Comment by eadventurous** [comment]
> Egor Ivkov, [06.07.21 13:10]
> [In reply to Ivan Rybin]
> Ok, let's log drop due to TTL for now
> 
> Egor Ivkov, [06.07.21 13:10]
> With a warn log level

@rkharisov can you do this instead? This will not require any additional fields. For client side testing it is possible to just set small ttl.

**Comment by rkharisov** [comment]
> > Egor Ivkov, [06.07.21 13:10]
> > [In reply to Ivan Rybin]
> > Ok, let's log drop due to TTL for now
> > Egor Ivkov, [06.07.21 13:10]
> > With a warn log level
> 
> @rkharisov can you do this instead? This will not require any additional fields. For client side testing it is possible to just set small ttl.

will do

**Review by i1i1 — APPROVED** [review]

**Review by eadventurous — APPROVED** [review]

**Pull request merged**